core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -2461,6 +2474,8 @@ private[spark] class DAGScheduler(
       clearCacheLocs()
     }
```

Comment:
We need to `removeShufflePushMergerLocation` even when `execId == SHUFFLE_MERGER_IDENTIFIER` (makes sense for rest of the `if` block to be within  `! isShuffleMerger` check; namely updating `executorFailureEpoch`, `removeExecutor` and `clearCacheLocs`)

Looks like we dont have a test to verify `removeShufflePushMergerLocation` is called ?
If not, we should add one.  +CC @otterc

Reply:
I've updated it to `removeShufflePushMergerLocation` even when `execId == SHUFFLE_MERGER_IDENTIFIER`. I also added a unit test to make sure that the method is being called.